FilterComplexCore.js

```diff
@@ -32,6 +32,31 @@ function getFieldVal(rowData, field) {
    return value;
 }
 
+function getConnectFieldValue(rowData, field) {
+   let connectedVal = "";
+
+   if (rowData) {
+      let relationName = field.relationName();
+      if (rowData[relationName]) {
+         connectedVal =
+            (field.indexField
+               ? rowData[relationName][field.indexField.columnName]
+               : false) || // custom index
```

Comment:
I think it might be better if we use ?? instead of ||

Reply:
`index ?? index2 ?? id ?? value` might be better. I think.

Reply:
Honestly I just borrowed this code from RowFilterCore https://github.com/digi-serve/appbuilder_class_core/blob/master/RowFilterCore.js#L456

Reply:
Okay I tried changing the `||` to `??` in the suggested areas of this file but the updates to the datatable I was trying to fix stopped working again. Should we keep it how we used to do it in RowFilterCore for now?

Reply:
It is interesting. Could you tell me how to reproduce it ?

Reply:
` (field.indexField
           ? rowData[relationName][field.indexField.columnName]
           : null) ?? // custom index
        (field.indexField2
           ? rowData[relationName][field.indexField2.columnName]
           : null) ?? // custom index 2
        rowData[relationName].id ??
        rowData[relationName]; `
        
Hope this code is able to solve

Reply:
Ah yes...I see when you change the `false` to `null` this now works...but is there a reason why we need to change the `false` to `null` just to make it work? Could we not just keep the `false` `||` combination? I am seeing that we are moving to using `??` in many places but often it just requires we add or change more code and I don't see the benefit. I am probably just unaware of something but the code seems to be working not without making these changes.

Reply:
It would more clear what the code do.

Previously, we used `||` because ES5 does not support `??`

Reply:
It would be more clear what the code do. (document itself)

Previously, we used `||` because ES5 does not support `??`